servers/rendering/rendering_device.cpp

```diff
@@ -7287,7 +7287,10 @@ void RenderingDevice::_set_max_fps(int p_max_fps) {
 
 RenderingDevice *RenderingDevice::create_local_device() {
 	RenderingDevice *rd = memnew(RenderingDevice);
-	rd->initialize(context);
+	if (rd->initialize(context) != OK) {
```

Comment:
I should note that this function is bound. So after this PR, users will no longer need to free a failed `create_local_device`. But it's probably correct to do this change because otherwise, it would be impossible to detect this failure...